x/pool-incentives/spec/03_gov.md

```diff
@@ -0,0 +1,71 @@
+<!--
+order: 3
+-->
+
+# Gov
+
+`Pool Incentives` module takes the uses the values set at genesis or values added by chain governance to distribute part of the inflation minted by the mint module to specified pots.
+
+```go
+type DistrInfo struct {
+	TotalWeight github_com_cosmos_cosmos_sdk_types.Int 
+	Records     []DistrRecord                          
+}
+
+type DistrRecord struct {
+	PotId  uint64                                 
+	Weight github_com_cosmos_cosmos_sdk_types.Int 
+}
+```
+`DistrInfo` internally manages the `DistrRecord` and total weight of all `DistrRecord`. Governance can't modify DistrInfo. It can only add/remove/edit the `DistrRecord`.
+
+### AddPoolIncentivesProposal
```

Comment:
Why don't we just merge all three of these governance proposals in to one, that's like `UpdatePoolIncentives`?

And it just overwrites all the DistrRecords in state with those in the new proposal?  Instead of needing multiple proposals to add and remove?

Reply:
With this, we could also have it so it always sums to <= 1.  And if the records sum to < 1, the excess should be given to the community pool

Reply:
I think this makes sense because in a way, I think governance should treat paying LP rewards as paying from the community pool.

Reply:
Yeah, fair point.  The number of incentivized pools in Sushiswap for example, is close to 150, which is probably pushing it.


In that case, could we make an explicit special DistrRecord that's meant specifically for the community pool?  That way governance can still choose to not spend all immediately, and instead direct a portion of newly minted coins to the community pool.

Reply:
We can also still have a unify into an UpdatePoolIncentives proposal regardless, right?

It could just be a list of `(potID, newWeight)`s

And if newWeight is 0, it removes the pot.  If the potID doesn't exist in the incentivized pool list, it adds it.

Reply:
Sorry I missed this thread. 

Changing every pool weight is peanuts work at `150` pools, computers are fast. Remember, we do lots of decimal operations for every validator per block for F1, work in slashing (large state loads to keep track of bit arrays), and staking. There are 125 validators on the hub, and this is not an issue.

I think we should still enforce that it sums to 1. Re-evaluating the weights is imo easy. 

I'm also pretty unconcerned about uncharged gas fees for governance proposals. The worst case here is that a block takes 100 extra miliseconds after 2/3rds of the community voted on a change, and even that is a huge stretch imo.

Reply:
@ValarDragon do you not think the size of the proposal itself is really big?  Like a list of say 300 potsIDs + weights?

Reply:
Wasn't thinking of the data size, now I agree. pot ID + weight, generously would take 100 bytes. so 1000 IDs would be 100kb, which is still well under our block limit. Its a pain to edit that json / verify it though, so I'm now in agreement with giving each pool a weight, and specifying a total weight

Reply:
I think for launch though, listing everything in a proposal suffices. We can make this flow better #postlaunch imo.

Reply:
This is done by UpdateDistrRecords and closing the thread